### PR TITLE
[REF] Clarify not-so-great code in the Group BAO

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Adds comments and a bit of tidying to some problematic code

Before
----------------------------------------
Problematic code in Group BAO; can only add parents but cannot remove them.

After
----------------------------------------
Better documented problematic code

### CODE REVIEW

File: settings/Multisite.setting.php [hunk 1]
```diff
@@ -20,6 +20,7 @@
  */
 
 return [
+  // FIXME: This is arguably the worst name for a setting ever
   'is_enabled' => [
```
Comment (colemanw) [inline settings/Multisite.setting.php:24]: @eileenmcnaughton maybe we could add a way to rename and alias settings? Probably trigger a deprecation when accessed by the old name.

**Comment by colemanw** [comment]
Thanks @mattwire!

File: settings/Multisite.setting.php (hunk 1 above)
Comment (totten) [reply settings/Multisite.setting.php:24]: I believe Eileen did a rename/reformat a few years back for a setting called `contribution_invoice_settings` (which was apparently a pretty gnarly setting). A straight-up rename should be simpler, but the important thing is that it led to the methods `SettingsBag::updateVirtual()` and `SettingsBag::createVirtual()`. That would probably be the easiest place to inject an alias mechanism.

Comment (colemanw) [reply settings/Multisite.setting.php:24]: @eileenmcnaughton I feel like its time to add noisy deprecation to `contribution_invoice_settings`

**Comment by samuelsov** [comment]
If we remove the parents in the CRM_Contact_BAO_Group::create function, I think this code will be redundant : https://github.com/civicrm/civicrm-core/blob/master/CRM/Group/Form/Edit.php#L372

**Comment by colemanw** [comment]
@samuelsov yes, that's true. The code is all a bit complicated but please feel free to try a PR.

**Comment by samuelsov** [comment]
@colemanw work done here if you want to review -> https://github.com/civicrm/civicrm-core/pull/26200

**Pull request merged**